cookbook/upgrade/major_version.rst

```diff
@@ -0,0 +1,114 @@
+.. index::
+    single: Upgrading; Major Version
+
+Upgrading a Major Version (e.g. 2.7.0 to 3.0.0)
+===============================================
+
+Every few years, Symfony releases a new major version release (the first number
+changes). These releases are the trickiest to upgrade, as they are allowed to
+contain BC breaks. However, Symfony tries to make this upgrade process as
+smooth as possible.
+
+This means that you can update most of your code before the major release is
+actually released. This is called making your code *future compatible*.
+
+There are a couple of steps to upgrading a major version:
+
+#. :ref:`Make your code deprecation free <upgrade-major-symfony-deprecations>`;
+#. :ref:`Update to the new major version via Composer <upgrade-major-symfony-composer>`.
+#. :ref:`Update your code to work with the new version <upgrade-major-symfony-after>`
+
+.. _upgrade-major-symfony-deprecations:
+
+1) Make your Code Deprecation Free
+----------------------------------
+
+During the lifecycle of a major release, new features are added and method
+signatures and public API usages are changed. However, minor versions should
+not contain any backwards compatibility changes. It is made sure that there is
+a so-called *backwards compatibility layer* (or BC layer). This means that the
+old API will still work, while the new feature is used internally. This BC
+layer is then marked as *deprecated*, indicating that it will be
+removed/changed in the future.
```

Comment:
I think this is more technical than we need to be. The important thing is that your old code still works, but is deprecated. Perhaps (starting right after "compatibility changes.")

```rst
To accomplish this, the "old" (e.g. functions, classes, etc) code still works, but is marked
as *deprecated*, indicating that it will be removed/changed in the future and that you
should stop using it.
```